includes/wc-core-functions.php

```diff
@@ -2273,3 +2273,13 @@ function wc_load_cart() {
 	WC()->initialize_session();
 	WC()->initialize_cart();
 }
+
+/**
+ * Test whether the context of execution comes from async action scheduler.
+ *
+ * @since 4.0.0
+ * @return bool
+ */
+function wc_is_running_from_async_action_scheduler() {
+	return isset( $_REQUEST['action'] ) && 'as_async_request_queue_runner' === $_REQUEST['action'];
```

Comment:
I wonder if this is the best way to check whether AS is running or not? but maybe we can fix this in a later PR, cc @rrennick

Reply:
👍 Yup, I think this needs a revisit. Maybe even to check for cron actions? More investigation is needed.